src/Base.cpp

```diff
@@ -36,6 +36,11 @@ int Base::getLife()
     return this->currentLife;
 }
 
+int Base::healLife(int addLife)
+{
+    return this->currentLife += addLife;
```

Comment:
também seria bom colocar um limite de vida máximo pra base e adicionar um caso aqui que se a regeneração for deixar a base com uma vida além da máxima permitida, não curar.